### ISSUE TITLE
Crash clicking on draw call in captured frame.

### ISSUE THREAD

**Issue by Lawrencemm** [baldurk/renderdoc#981]
Trying to debug a small app drawing a triangle.
Triangle is drawn on Windows platform, but is not drawn on linux.
After capturing a frame, I click on the draw call and the renderdoc insta-crashes.

If you wish to repro, pull https://github.com/lawrencem99/workspace 
init the submodules and the build should be easy with cmake (I use clion).
begin capture with renderdoc and press the up or down arrow on the keyboard to render a frame.

Here's the report from the crash dump, including stack trace:
https://bugzilla.redhat.com/show_bug.cgi?id=1576630

**Comment by Lawrencemm** [comment]
Having fixed the issue in my program, I'm finding renderdoc still crashes on linux.

It's lucky I could reproduce my program's bug by using a different compiler on windows (thanks cmake) because renderdoc was essential in solving the problem.

**Comment by baldurk** [comment]
Hi, I've been trying to figure out how to build your program but without much success unfortunately, all of the compilers I've tried have failed in one way or another - VS2015/2017 on windows and gcc 5/6/7/8 + clang 4/5/6 on linux.

Can you tell me what compiler you're using, or which ones you expect to work?

**Comment by Lawrencemm** [comment]
My submodule URLs were ssh links.
I pushed an update to make them work with https clones.

Tried a clone in a fresh fedora 28 install, successfully built with GCC 8.

dependencies are 
vulkan-devel wayland-devel libxcb-devel xcb-util-devel

For Fedora, or their analogues on other distros.

**Comment by baldurk** [comment]
The ssh links were fine, I clone via ssh as well. It was a build error I ran into:

```
[ 81%] Building CXX object lmlib/CMakeFiles/lmlib.dir/src/graphics_vulkan/renderer.cpp.o
cd /home/baldurk/issue981/build/lmlib && /usr/bin/g++-8  -DVK_PROTOTYPES -DVK_USE_PLATFORM_WAYLAND_KHR -DVK_USE_PLATFORM_XCB_KHR -I/home/baldurk/issue981/embed-resource -I/home/baldurk/issue981/lmlib/SYSTEM -I/home/baldurk/issue981/eigen -I/home/baldurk/issue981/lmlib/include/lmlib -I/home/baldurk/issue981/lmlib/include -I/home/baldurk/issue981/tbb/include -I/home/baldurk/issue981/freetype2/include -I/home/baldurk/issue981/openmesh/src  -std=c++17 -Wno-error -g   -w -o CMakeFiles/lmlib.dir/src/graphics_vulkan/renderer.cpp.o -c /home/baldurk/issue981/lmlib/src/graphics_vulkan/renderer.cpp
In file included from /home/baldurk/issue981/lmlib/include/lmlib/graphics_vulkan/exceptions.h:4,
                 from /home/baldurk/issue981/lmlib/src/graphics_vulkan/renderer.cpp:1:
/usr/include/vulkan/vulkan.hpp: In instantiation of ‘vk::ResultValueType<vk::UniqueHandle<vk::Device> >::type vk::PhysicalDevice::createDeviceUnique(const vk::DeviceCreateInfo&, vk::Optional<const vk::AllocationCallbacks>, const Dispatch&) const [with Dispatch = vk::DispatchLoaderStatic; vk::ResultValueType<vk::UniqueHandle<vk::Device> >::type = vk::UniqueHandle<vk::Device>]’:
/home/baldurk/issue981/lmlib/src/graphics_vulkan/renderer.cpp:189:27:   required from here
/usr/include/vulkan/vulkan.hpp:35431:29: error: no matching function for call to ‘vk::ObjectDestroy<vk::NoParent>::ObjectDestroy(const vk::PhysicalDevice&, vk::Optional<const vk::AllocationCallbacks>&)’
     ObjectDestroy<NoParent> deleter( *this, allocator );
                             ^~~~~~~
In file included from /home/baldurk/issue981/lmlib/include/lmlib/graphics_vulkan/exceptions.h:4,
                 from /home/baldurk/issue981/lmlib/src/graphics_vulkan/renderer.cpp:1:
/usr/include/vulkan/vulkan.hpp:1014:5: note: candidate: ‘vk::ObjectDestroy<vk::NoParent>::ObjectDestroy(vk::Optional<const vk::AllocationCallbacks>)’
     ObjectDestroy( Optional<const AllocationCallbacks> allocator = nullptr )
     ^~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:1014:5: note:   candidate expects 1 argument, 2 provided
/usr/include/vulkan/vulkan.hpp:1011:9: note: candidate: ‘constexpr vk::ObjectDestroy<vk::NoParent>::ObjectDestroy(const vk::ObjectDestroy<vk::NoParent>&)’
   class ObjectDestroy<NoParent>
         ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan.hpp:1011:9: note:   candidate expects 1 argument, 2 provided
/usr/include/vulkan/vulkan.hpp:1011:9: note: candidate: ‘constexpr vk::ObjectDestroy<vk::NoParent>::ObjectDestroy(vk::ObjectDestroy<vk::NoParent>&&)’
/usr/include/vulkan/vulkan.hpp:1011:9: note:   candidate expects 1 argument, 2 provided
lmlib/CMakeFiles/lmlib.dir/build.make:466: recipe for target 'lmlib/CMakeFiles/lmlib.dir/src/graphics_vulkan/renderer.cpp.o' failed
make[2]: *** [lmlib/CMakeFiles/lmlib.dir/src/graphics_vulkan/renderer.cpp.o] Error 1
CMakeFiles/Makefile2:2358: recipe for target 'lmlib/CMakeFiles/lmlib.dir/all' failed
make[1]: *** [lmlib/CMakeFiles/lmlib.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
```

This is the gcc 8 version I'm using: `g++-8 (Ubuntu 8-20180414-1ubuntu2) 8.0.1 20180414 (experimental) [trunk revision 259383]`

Also FYI I get an error about `<xcb/xinput.h>` being missing, though I can't tell why. It's maybe only available on fedora or something, on google it seems like maybe Ubuntu's version of xcb isn't built with xinput support? But just commenting out that include didn't break anything so I assume it's not needed.

**Comment by Lawrencemm** [comment]
I believe that's an error from a previous version of the Vulkan sdk, vulkan hpp was broken in the first 1.1 sdks.

which version are you on? I had to wait for 1.1.73 before I could use the Vulkan 1.1 sdk versions.

**Comment by baldurk** [comment]
You're right, the latest version available in ubuntu's stable release is 1.1.70. Since it doesn't have many external dependencies I installed 1.1.73 from the next release of ubuntu, which should probably be fine...

That got a little further but it now fails with this:

```
[ 98%] Building CXX object lmlib/CMakeFiles/test_lmlib_linux.dir/test/src/xcb.cpp.o
cd /home/baldurk/issue981/build/lmlib && /usr/bin/g++-8  -DVK_PROTOTYPES -DVK_USE_PLATFORM_WAYLAND_KHR -DVK_USE_PLATFORM_XCB_KHR -I/home/baldurk/issue981/embed-resource -I/home/baldurk/issue981/lmlib/SYSTEM -I/home/baldurk/issue981/eigen -I/home/baldurk/issue981/lmlib/include -I/home/baldurk/issue981/tbb/include -I/home/baldurk/issue981/freetype2/include -I/home/baldurk/issue981/openmesh/src -isystem /home/baldurk/issue981/googletest/googletest/include  -std=c++17 -Wno-error -g   -w -o CMakeFiles/test_lmlib_linux.dir/test/src/xcb.cpp.o -c /home/baldurk/issue981/lmlib/test/src/xcb.cpp
/home/baldurk/issue981/lmlib/test/src/xcb.cpp: In member function ‘virtual void xcb_draw_with_vulkan_Test::TestBody()’:
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:36:12: error: aggregate ‘lm::graphics_vulkan::Renderer renderer’ has incomplete type and cannot be defined
   Renderer renderer;
            ^~~~~~~~
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:37:3: error: ‘Stage’ was not declared in this scope
   Stage stage;
   ^~~~~
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:37:3: note: suggested alternative: ‘rusage’
   Stage stage;
   ^~~~~
   rusage
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:38:3: error: ‘Scene’ was not declared in this scope
   Scene scene;
   ^~~~~
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:40:3: error: ‘stage’ was not declared in this scope
   stage.init(renderer, window);
   ^~~~~
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:40:3: note: suggested alternative: ‘stat’
   stage.init(renderer, window);
   ^~~~~
   stat
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:41:3: error: ‘scene’ was not declared in this scope
   scene.background_colour(std::array<float, 4>{0.2f, 0.2f, 0.2f, 1.f});
   ^~~~~
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:41:3: note: suggested alternative: ‘scanf’
   scene.background_colour(std::array<float, 4>{0.2f, 0.2f, 0.2f, 1.f});
   ^~~~~
   scanf
/home/baldurk/issue981/lmlib/test/src/xcb.cpp:48:4: error: no matching function for call to ‘lm::linux::XcbWindow::on_key_down(xcb_draw_with_vulkan_Test::TestBody()::<lambda(lm::pc_platform::IWindow&)>)’
   });
    ^
In file included from /home/baldurk/issue981/lmlib/include/lmlib/linux/xcb.h:13,
                 from /home/baldurk/issue981/lmlib/test/src/xcb.cpp:2:
/home/baldurk/issue981/lmlib/include/lmlib/pc_platform/pc_platform.h:72:8: note: candidate: ‘void lm::pc_platform::IWindow::on_key_down(lm::pc_platform::IWindow::KeyDownHandler)’
   void on_key_down(KeyDownHandler handler)
        ^~~~~~~~~~~
/home/baldurk/issue981/lmlib/include/lmlib/pc_platform/pc_platform.h:72:8: note:   no known conversion for argument 1 from ‘xcb_draw_with_vulkan_Test::TestBody()::<lambda(lm::pc_platform::IWindow&)>’ to ‘lm::pc_platform::IWindow::KeyDownHandler’ {aka ‘std::function<void(lm::pc_platform::IWindow&, lm::pc_platform::Keycode)>’}
lmlib/CMakeFiles/test_lmlib_linux.dir/build.make:62: recipe for target 'lmlib/CMakeFiles/test_lmlib_linux.dir/test/src/xcb.cpp.o' failed
make[2]: *** [lmlib/CMakeFiles/test_lmlib_linux.dir/test/src/xcb.cpp.o] Error 1
CMakeFiles/Makefile2:2440: recipe for target 'lmlib/CMakeFiles/test_lmlib_linux.dir/all' failed
make[1]: *** [lmlib/CMakeFiles/test_lmlib_linux.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
```

And very similar errors in `wayland.cpp`.

**Comment by Lawrencemm** [comment]
Build target "lmeditor" then run "<build_root>/lmeditor/Editor"

**Comment by baldurk** [comment]
Thanks, I got it to run. Unfortunately after all that I _think_ this is a known bug from v1.0 that didn't support mesh output fetching with vertex input formats that aren't supported as texel buffers - e.g. in this case your position stream is `R32G32B32_SFLOAT` which is sometimes only supported for vertex buffer access.

I can't confirm since on my hardware `R32G32B32_SFLOAT` is supported for both, so it works for me. It's been fixed since v1.0, so can you try to run a recent nightly build or compile from latest source and test?

**Comment by Lawrencemm** [comment]
Works with nightly.

Will try that first next time. Thanks for the help.
